server/helpers/pagination.js

```diff
@@ -1,11 +1,11 @@
 export const generatePagination = (page, limit, total) => {
   const currentPage = parseInt(page, 10) < 1 ? 1 : parseInt(page, 10);
   return {
-    currentPage,
+    currentPage: limit === 0 ? 1 : currentPage,
```

Comment:
I think it is better to use a variable for this instead of the expression.

```suggestion
    currentPage:  removeLimit ? 1 : currentPage,
```

and somewhere above, that variable is defined as

```js
   const removeLimit = limit === 0;
```